cmd/scollector/collectors/apache_mod_info.go

```diff
@@ -0,0 +1,121 @@
+package collectors
+
+import (
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"golang.org/x/net/html"
+
+	"bosun.org/metadata"
+	"bosun.org/opentsdb"
+)
+
+func init() {
+	collectors = append(collectors, &IntervalCollector{F: c_apache_mod_info, Interval: time.Minute * 30})
+}
+
+// Extract from HTML recieved from http://127.0.0.1/server-info?server
+// Find the line that looks like:
+// "Timeouts: connection: 60 keep-alive: 15"
+// Returns the two values contained therein.
+
+func extract_timeouts(doc *html.Node) (int, int) {
+
+	// Walk the node tree.
+	// If the node is a "dt",
+	//     walk its children until we find CONTENTS "Timeouts:"
+	//         At which point we walk those children to find "connection".
+	// Else, keep walking.
+	//     Then continue walking Children to find CONTENTS startswith "connection":
+	//     Return contents
+
+	var value string
+	var search_children func(*html.Node) string
+	search_children = func(n *html.Node) string {
+		if n.Type == html.TextNode {
+			if strings.HasPrefix(n.Data, "connection:") {
+				value = n.Data
+				return value
+			}
+		}
+		for c := n.FirstChild; c != nil; c = c.NextSibling {
+			if value = search_children(c); value != "" {
+				return value
+			}
+		}
+		return value
+	}
+
+	var walk_subtree func(*html.Node) string
+	walk_subtree = func(n *html.Node) string {
+		if n.Type == html.ElementNode && n.Data == "dt" {
+			if val := search_children(n); val != "" {
+				return val
+			}
+		} else {
+			for c := n.FirstChild; c != nil; c = c.NextSibling {
+				if val := walk_subtree(c); val != "" {
+					return val
+				}
+			}
+		}
+		return ""
+	}
+
+	original := walk_subtree(doc)
+	parts := strings.Fields(original)
+	if (parts[0] != "connection:") || (parts[2] != "keep-alive:") {
+		log.Fatal("Format of connection: / keep-alive: line changed")
+	}
+
+	c, _ := strconv.Atoi(parts[1])
```

Comment:
strconv.Atoi returns an `int`, which can be as small as 32bits. If this number can ever not fit in a 32-bit signed int, switch to strconv.ParseInt, which returns an int64.